internal/ent/hooks/trustcenter.go

```diff
@@ -294,21 +347,53 @@ func HookTrustCenterUpdate() ent.Hook {
 				return v, err
 			}
 
+			if customDomainCleared || (mutationCustomDomainIDExists && mutationCustomDomainID == "") {
+				if previousCustomDomainID != nil && previousPirschDomainID != "" {
+					if err := enqueueJob(ctx, m.Job, jobspec.DeletePirschDomainArgs{
+						PirschDomainID: previousPirschDomainID,
+					}, nil); err != nil {
+						return nil, err
+					}
+				}
+
+				if previousCustomDomainID != nil && m.Job != nil {
+					if cd, err := m.Client().CustomDomain.Get(ctx, *previousCustomDomainID); err == nil && cd.CnameRecord != "" {
+						if err := enqueueJob(ctx, m.Job, jobspec.ClearTrustCenterCacheArgs{
+							CustomDomain: cd.CnameRecord,
+						}, nil); err != nil {
+							return nil, err
+						}
+					}
+				}
+
+				return v, nil
+			}
+
 			if mutationCustomDomainIDExists && previousCustomDomainID == nil && mutationCustomDomainID != "" {
-				if _, err := m.Job.Insert(ctx, jobspec.CreatePirschDomainArgs{
+				if err := enqueueJob(ctx, m.Job, jobspec.CreatePirschDomainArgs{
 					TrustCenterID: tcID,
 				}, nil); err != nil {
 					return nil, err
 				}
-			} else if mutationCustomDomainIDExists && previousCustomDomainID != nil && mutationCustomDomainID != *previousCustomDomainID {
-				if _, err := m.Job.Insert(ctx, jobspec.UpdatePirschDomainArgs{
+			} else if mutationCustomDomainIDExists && previousCustomDomainID != nil && mutationCustomDomainID != "" && mutationCustomDomainID != *previousCustomDomainID {
+				if err := enqueueJob(ctx, m.Job, jobspec.UpdatePirschDomainArgs{
 					TrustCenterID: tcID,
 				}, nil); err != nil {
 					return nil, err
 				}
+
+				if m.Job != nil {
+					if cd, err := m.Client().CustomDomain.Get(ctx, *previousCustomDomainID); err == nil && cd.CnameRecord != "" {
+						if err := enqueueJob(ctx, m.Job, jobspec.ClearTrustCenterCacheArgs{
+							CustomDomain: cd.CnameRecord,
+						}, nil); err != nil {
+							return nil, err
+						}
+					}
+				}
```

Comment:
Same here. If can see you want to avoid the queries if there is no job client to have this check for efficiency but if that were the reason, I would do it once before all of these and return early and not before some of them.  Since `enqueueJob` checks this, its not needed here to avoid nil pointers. 
```suggestion
			if cd, err := m.Client().CustomDomain.Get(ctx, *previousCustomDomainID); err == nil && cd.CnameRecord != "" {
				if err := enqueueJob(ctx, m.Job, jobspec.ClearTrustCenterCacheArgs{
					CustomDomain: cd.CnameRecord,
				}, nil); err != nil {
					return nil, err
				}
			}
```